crypto/ecies/ecies.go

```diff
@@ -138,57 +139,39 @@ func (prv *PrivateKey) GenerateShared(pub *PublicKey, skLen, macLen int) (sk []b
 }
 
 var (
-	ErrKeyDataTooLong = fmt.Errorf("ecies: can't supply requested key data")
 	ErrSharedTooLong  = fmt.Errorf("ecies: shared secret is too long")
 	ErrInvalidMessage = fmt.Errorf("ecies: invalid message")
 )
 
-var (
-	big2To32   = new(big.Int).Exp(big.NewInt(2), big.NewInt(32), nil)
-	big2To32M1 = new(big.Int).Sub(big2To32, big.NewInt(1))
-)
-
-func incCounter(ctr []byte) {
-	if ctr[3]++; ctr[3] != 0 {
-		return
-	}
-	if ctr[2]++; ctr[2] != 0 {
-		return
-	}
-	if ctr[1]++; ctr[1] != 0 {
-		return
-	}
-	if ctr[0]++; ctr[0] != 0 {
-		return
-	}
-}
-
 // NIST SP 800-56 Concatenation Key Derivation Function (see section 5.8.1).
-func concatKDF(hash hash.Hash, z, s1 []byte, kdLen int) (k []byte, err error) {
-	if s1 == nil {
-		s1 = make([]byte, 0)
-	}
-
-	reps := ((kdLen + 7) * 8) / (hash.BlockSize() * 8)
-	if big.NewInt(int64(reps)).Cmp(big2To32M1) > 0 {
-		fmt.Println(big2To32M1)
-		return nil, ErrKeyDataTooLong
-	}
-
-	counter := []byte{0, 0, 0, 1}
-	k = make([]byte, 0)
-
-	for i := 0; i <= reps; i++ {
-		hash.Write(counter)
+func concatKDF(hash hash.Hash, z, s1 []byte, kdLen int) []byte {
```

Comment:
Previously, this method could return an error, `ErrKeyDataTooLong`. So what would happen _now_ when the same input that would trigger `ErrKeyDataTooLong` is fed into this one? Is that error only an implementation flaw that went away?

Reply:
The error would be returned when the requested amount of key data was larger than `2^32-1`. This error could never actually happen because the concatKDF is only used internally in crypto/ecies and we always request exactly 32 bytes.

Reply:
That sounds odd. So you're saying that if `kdLen > 2^32-1`, then the result is not correct, but it doesn't matter because we always have it at 32? 
Why don't we hardcode it at 32 then, and drop the params? Alternatively, set kdLen to `uint8` or something?

Reply:
I think it is unlikely anyone will ever request gigabytes of secret key data from this function. If `kdLen` is very large, the result isn't necessarily incorrect, it just doesn't correspond to the NIST SP 800-56 spec.